test/defaultParseAndHandleMessage-test.js

```diff
@@ -5,154 +5,112 @@ import assert from 'assert'
 
 import defaultParseAndHandleMessage from '../src/defaultParseAndHandleMessage'
 
+const verifyMessageHandling = (handleError, handleMessage, content) => {
+  const message = { content }
+
+  let handleCount = 0
+  const _handleMessage = async (data) => {
+    handleCount++
+    return handleMessage(data)
+  }
+
+  let ackCount = 0
+  const ack = msg => {
+    ackCount++
+    assert.strictEqual(message, msg)
+  }
+
+  const nack = () => {
+    throw new Error('should not nack')
+  }
+
+  let handleErrorCount = 0
+  const _handleError = async (e, msg) => {
+    handleErrorCount++
+    assert.strictEqual(message, msg)
+    return handleError(e, msg)
+  }
+
+  const parseAndHandle = defaultParseAndHandleMessage(_handleError, _handleMessage)
+
+  return parseAndHandle(ack, nack, message)
+    .then(
+      () => ({ ackCount, handleCount, handleErrorCount }),
+      () => Promise.reject({ ackCount, handleCount, handleErrorCount })
+    )
+}
+
+const verifyCounts = (expectedAckCount, expectedHandleCount, expectedHandleErrorCount) =>
+  ({ ackCount, handleCount, handleErrorCount }) => {
+    assert.equal(expectedAckCount, ackCount)
+    assert.equal(expectedHandleCount, handleCount)
+    assert.equal(expectedHandleErrorCount, handleErrorCount)
+  }
+
 describe('defaultParseAndHandleMessage', () => {
   // TODO: This *should nack*, but see comments in parseAndHandleMessage
   // for why it (temporarily) acks instead.
   it('should call handleError and ack if message handling fails', () => {
     const content = JSON.stringify({ foo: 'bar' })
-    const message = { content }
 
-    let handleCount = 0
-    const handleMessage = async (data) => {
-      handleCount++
-      assert.equal(content, JSON.stringify(data))
-      throw new Error()
-    }
-
-    let ackCount = 0
-    const ack = msg => {
-      ackCount++
-      assert.strictEqual(message, msg)
-    }
-
-    const nack = () => {
-      throw new Error('should not nack')
-    }
-
-    let handleErrorCount = 0
     const handleError = async (error) => {
```

Comment:
I think it's sufficient to assert [here](https://github.com/nowait/amqp/blob/d72ef659b56b7db24c6ef886258287b835cac0f2/test/defaultParseAndHandleMessage-test.js#L30).